crates/ere-cli/src/main.rs

```diff
@@ -0,0 +1,213 @@
+            + cfg!(feature = "risc0") as u8
+            + cfg!(feature = "sp1") as u8
+            + cfg!(feature = "zisk") as u8)
+            == 1,
+        "Exactly one zkVM backend feature must be enabled"
+    );
+};
+
+#[derive(Parser)]
+#[command(author, version)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Compile a guest program
+    Compile {
+        /// Path to the base directory (workspace root)
+        #[arg(long)]
+        mount_directory: PathBuf,
+        /// Relative path from `mount_directory` to the guest program
+        #[arg(long)]
+        guest_relative: PathBuf,
+        /// Path where the compiled program will be written
+        #[arg(long)]
+        program_path: PathBuf,
+    },
+    /// Execute a compiled program
+    Execute {
+        /// Path to the compiled program
+        #[arg(long)]
+        program_path: PathBuf,
+        /// Path to the serialized input bytes file
+        #[arg(long)]
+        input_path: PathBuf,
+        /// Path where the execution report will be written
+        #[arg(long)]
+        report_path: PathBuf,
+        // Prover resource type
+        #[command(subcommand)]
+        resource: ProverResourceType,
+    },
+    /// Prove execution of a compiled program
+    Prove {
+        /// Path to the compiled program
+        #[arg(long)]
+        program_path: PathBuf,
+        /// Path to the serialized input bytes file
+        #[arg(long)]
+        input_path: PathBuf,
+        /// Path where the proof will be written
+        #[arg(long)]
+        proof_path: PathBuf,
+        /// Path where the report will be written
+        #[arg(long)]
+        report_path: PathBuf,
+        // Prover resource type
+        #[command(subcommand)]
+        resource: ProverResourceType,
+    },
+    /// Verify execution of a compiled program
+    Verify {
+        /// Path to the compiled program
+        #[arg(long)]
+        program_path: PathBuf,
+        /// Path to the proof
+        #[arg(long)]
+        proof_path: PathBuf,
+    },
+}
+
+fn main() -> Result<(), Error> {
+    let args = Cli::parse();
+
+    match args.command {
+        Commands::Compile {
+            mount_directory,
+            guest_relative,
+            program_path,
+        } => compile(mount_directory, guest_relative, program_path),
+        Commands::Prove {
+            program_path,
+            input_path,
+            proof_path,
+            report_path,
+            resource,
+        } => prove(program_path, resource, input_path, proof_path, report_path),
+        Commands::Execute {
+            program_path,
+            input_path,
+            report_path,
+            resource,
+        } => execute(program_path, resource, input_path, report_path),
+        Commands::Verify {
+            program_path,
+            proof_path,
+        } => verify(program_path, proof_path),
+    }
+}
+
+fn compile(
```

Comment:
Re https://github.com/eth-act/ere/pull/66#discussion_r2242911606 I think the macros seems a bit unnecessary so I remove it

Re https://github.com/eth-act/ere/pull/66#discussion_r2242917143 Previously there are some program sereizlied and some not, the reason is because some program is already `Vec<u8>` so I wanted to avoid double serialization, but it seems to be a footgun so I just serialize every program now.